src/edu/washington/escience/myria/TupleWriter.java

```diff
@@ -20,7 +22,15 @@
  * 
  * 
  */
-public interface TupleWriter {
+public interface TupleWriter extends Serializable {
+
+  /** Required for Java serialization. */
+  static final long serialVersionUID = 1L;
```

Comment:
I don't believe this will work--see https://docs.oracle.com/javase/8/docs/api/java/io/Serializable.html: 
> such declarations apply only to the immediately declaring class--serialVersionUID fields are not useful as inherited members.